### PR TITLE
Monster cameras, vision optimizations, and an implementation through Slime

### PR DESCRIPTION
**Pull request by MylieDaniels**
#### Summary
Features "Monsters can serve as cameras, adds Slimewave, optimizes seen_cache usage"

#### Purpose of change
Heavily inspired by @bombasticSlacks wonderful PR #55625, this adds the ability to use friendly monsters as an additional source of vision.  In addition, it adds a trait to Slime that allows for vision through Slimesprings (renamed to Slime Microbians) within 8 tiles, and through a new monster that cannot yet be created, the Slime Ocularium, within 32 tiles.  The Ocularium is not yet creatable because it is created as part of a WIP mutation rework- it is the reason I made this change, to be honest.
While testing, I found that seen_caches were used in excess, and optimized their usage, allowing for more accurate vision on short term activities and faster long activities.

#### Describe the solution
Vision can be cast from any friendly monster, specified by a "moncams" array in any trait/mutation.
The monster's daytime vision is the maximum view distance, and the moncams array specifies the range at which it can "transmit" to the avatar.
Vision will be cast only on the Z-level the monster is on- as such, a monster cam on a different Z-level does nothing when not using experimental Z-level, and will only cast on it's own level, even when using experimental Z-level vision.  This is because having monster cams cast full Z-level vision was found to be rather slow.
Vision from cameras is no longer limited to the player's maximum view distance, but is of course still limited to the reality bubble.
map::build_map_cache (called every turn) no longer rebuilds the seen cache, and instead, map::build_lightmap (a new function, called when player vision should update) handles that.

#### Testing
Mutated Slimewave, spawned Oculariums and Microbians and tested in a variety of situations, from vehicles to a limited usage in experimental Z-levels.
Tested long term resting and activities, found that incoming enemies were correctly detected, etc.

#### Additional context
image
I am on the second floor, in an enclosed room.  There are two Oculariums on the first floor, represented by green capital I's.

### CODE REVIEW

**Comment by Theawesomeboophis** [comment]
Slime cameras, something I never knew I wanted in a game until right now, this sounds awesome.

**Comment by MylieDaniels** [comment]
This ended up rather sprawling, touching many parts of vision code.  Not touching it again unless a bug is found, not even for the complete jsonification of sight_max that I will be doing in a different PR.

File: data/json/monsters/slimes.json [hunk 1]
```diff
@@ -174,8 +174,8 @@
   {
     "id": "mon_player_blob",
     "type": "MONSTER",
-    "name": { "str": "slimespring" },
-    "description": "A familiar-looking blob of goo.  It sprouts the occasional eyestalk.",
+    "name": { "str": "slime microbian" },
+    "description": "A small wriggling slime, designed as a first line of defense for the colony.",
```
Comment (kevingranade) [inline data/json/monsters/slimes.json:178]: Designed isn't really the right word, maybe just state what it's for.
```suggestion
    "description": "A small wriggling slime, a first line of defense for the colony.",
```

File: data/json/mutations/mutations.json [hunk 2]
```diff
@@ -5030,14 +5031,29 @@
...
     "passive_mods": { "per_mod": 5 }
   },
+  {
+    "type": "mutation",
+    "id": "SLIMEMIND",
+    "name": { "str": "Slimewave" },
+    "points": 10,
+    "visibility": 2,
+    "ugliness": 10,
+    "description": "You can communicate with your extant mass without physical contact, using electrical impulses to absorb knowledge from nearby friendly slime.",
```
Comment (kevingranade) [inline data/json/mutations/mutations.json:5049]: I'd say you don't know the mechanism, just that it happens.
```suggestion
    "description": "You can communicate with your extant mass without physical contact, absorbing knowledge from nearby friendly slime.",
```

**Pull request merged**